### DISCUSSION TITLE
config.json "include-tables" regex not working as expected

### DISCUSSION THREAD

**Discussion by will-scc** [linq2db/linq2db#4505] [Q&A]
I have a config file which looks like this (connection info removed):

```json
{
  "general": {
    "provider": "Oracle",
    "output": "Models",
    "overwrite": true
  },
  "schema": {
    "objects": [
      "table",
      "view",
      "foreign-key",
      "stored-procedure"
    ],
    "include-tables": [
      { "regex": "(?i)alerts" }
    ],
    "include-views": [
      { "regex": "(?i)failure_view" }
    ],
    "include-stored-procedures": [
      { "regex": "(?i)close_alert" }
    ]
  },
  "code": {
    "namespace": "Models"
  }
}
```

I'm trying to make it so that the list of objects specified under `include-tables`, `include-views`, and `include-stored-procedures` are not case sensitive. According to regex101 this should work: [https://regex101.com/r/A8RMvh/1](https://regex101.com/r/A8RMvh/1)

But when I run the scaffold command it seems to be matching other objects that it shouldn't.

What should the regex be to match the string completely, but be case insensitive?

**Comment by MaceWindu** [comment]
Any examples of wrong matches?

**Reply by will-scc** [reply]
Trying to scaffold our whole database throws another strange error (it doesn't like the database type NUMBER(22) apparently), so I have to specify a subset of the objects, with `include-tables`. 

If I just do `"include-tables": [ "alerts" ]` it will bring back the `alerts` table only, and correctly, but for some reason doing `"include-tables": [ { "regex": "(?i)alerts" } ]` it throws that other exception which we encountered when scaffolding all tables, which is what leads me to believe the regex is matching more than what I'm expecting. 

I hope that makes sense.
